activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -501,6 +510,21 @@ def combine_bind_parameters(
 
       private
 
+        def with_connection_verify(exception_class)
+          yield
+        rescue exception_class
+          raise_unless_verifying_connection
+          reconnect!
+          retry
+        ensure
+          @verify_connection = false
+        end
+
+        def raise_unless_verifying_connection
+          raise if !@verify_connection || active?
```

Comment:
could we ever get a `@verify_connection` not defined warning in here? as the `defer_verify` method first defines that ivar.
Also, I am not 100% what this flag is for. Can you expand?